### ISSUE TITLE
ICS27: Spec improvement suggestions 

### ISSUE THREAD

**Issue by andrey-kuprianov** [cosmos/ibc#618]
_Surfaced from @informalsystems audit of ICS27 InterchainAccounts specification at hash 058a3d68

Artifacts:

- [ICS27 Spec](https://github.com/cosmos/ibc/tree/058a3d68bfbf94029bf02fa16a6f0d33cfbaa5eb/spec/app/ics-027-interchain-accounts)

## [Synopsis.Motivation](https://github.com/cosmos/ibc/tree/058a3d68bfbf94029bf02fa16a6f0d33cfbaa5eb/spec/app/ics-027-interchain-accounts#motivation)

Explaining motivation via referring to Ethereum seems to be not the best possible way:
  - This gives the impression of Cosmos being inferior to its older and superior brother Ethereum
  - It doesn't convey the idea directly, especially for someone not familiar with Ethereum.
 
**Recommendation**: it seems better to explain the motivation taking one or two easy to understand use cases, and only after that referring to a similar mechanism on Ethereum.

## [Synopsis.Definitions](https://github.com/cosmos/ibc/tree/058a3d68bfbf94029bf02fa16a6f0d33cfbaa5eb/spec/app/ics-027-interchain-accounts#definitions)

The definition for "interchain account" mentions "host chain" and "controller chain" before they are introduced. 

**Recommendation**: Reorder the definitions as follows:
 - host chain
 - controller chain
 - interchain account
 - interchain account owner

## [Synopsis.Desired Properties](https://github.com/cosmos/ibc/tree/058a3d68bfbf94029bf02fa16a6f0d33cfbaa5eb/spec/app/ics-027-interchain-accounts#desired-properties)

"Permissionless" is only written down, but not explained.

**Recommendation**: Explain, at least shortly, what is understood under permissionless.

## [Technical Specification.Contract](https://github.com/cosmos/ibc/tree/058a3d68bfbf94029bf02fa16a6f0d33cfbaa5eb/spec/app/ics-027-interchain-accounts#contract)

The lines in the contract code snippet are extremely long (the longest is 230 characters); this prevents the code and comments from being efficiently read. Besides, there is not much actual *code* there: 90% is the text in comments describing in plain English what the functions are supposed to do.

**Recommendation**: Restructure the contract description:
 - Make each function a separate subsection with the spec part and the code part
 - Extract the English specification of a function (currently in comments) into the standard content of the .md file for the respective section
 - Keep in the code snippet for the respective function only its signature, and whatever can be expressed as a pseudo-code. 

## [Technical Specification.Contract.InitInterchainAccount](https://github.com/cosmos/ibc/tree/058a3d68bfbf94029bf02fa16a6f0d33cfbaa5eb/spec/app/ics-027-interchain-accounts#contract)

I believe in the below description the third line interrupts the sentence flow between the surrounding lines:

```go
// InitInterchainAccount is the entry point to registering an interchain account.
// It generates a new port identifier using the owner address, connection identifiers
// The port id will look like: ics27-1-{connection-number}-{counterparty-connection-number}-{owner-address}
// and counterparty connection identifier. It will bind to the port identifier and
// call 04-channel 'ChanOpenInit'. An error is returned if the port identifier is already in use
// An `OnChannelOpenInit` event is emitted which can be picked up by an offchain process such as a relayer
```

## Extract format specifications in a separate section

Descriptions of formats for ids, channel IDs, etc. are scattered throughout the spec. E.g. I have counted 4 occurrences of a port ID described in the following way:

``` 
The IBC portID will look like this:
  ics27-1-{connection-number}-{counterparty-connection-number}-{owner-address}
```

Doing it this way creates several problems:
 - it clutters the spec;
 - makes it hard to read, because each description needs to be parsed and compared to the same in other places;
 - it increases the likelihood of missing some of those places on subsequent spec modifications, thus making it inconsistent.

**Recommendation**: extract format specifications into a separate section; in all other places use internal references to this section.

**Comment by mpoke** [comment]
@AdityaSripal I see you addressed this issue in multiple commits. Should we close it?

**Comment by AdityaSripal** [comment]
Yes I addressed the points here in the current spec. Please reopen if there is anything missing with a comment on the specific points that need to be added.

Thanks!

**Comment by gjermundgaraba** [comment]
This issue was mistakenly reopened during some integration work, so closing again now.
